two_factor/views/core.py

```diff
@@ -548,8 +548,10 @@ def get_context_data(self, form, **kwargs):
             rawkey = unhexlify(key.encode('ascii'))
             b32key = b32encode(rawkey).decode('utf-8')
             self.request.session[self.session_key_name] = b32key
+            key = ' '.join(b32key[i:i+4] for i in range(0, len(b32key), 4))
```

Comment:
I don't like those spaces, what if the user needs to copy and paste the code into an app?

It might be possible to have the same effect without literal spaces via some HTML and CSS.

Reply:
They can use `|cut:" "` for that (on the other hand there are no methods to easily add spaces in template). Without this it is harder to separate them out. Most sites I have seen online display it with spaces. Sticking it together makes it harder to read.

Reply:
End users can't use `|cut:" "` and going to be pretty annoyed when they can't copy and paste that code into their app because of the spaces.

Reply:
What's wrong with a template tag that groups the key chars into 4s and wraps them in `<span>` tags? That way when the user selects them to copy and paste there are no spaces *and* we could use some CSS to add padding to make the code easier to read.

Reply:
I mean developers can just put `|cut:" "` to let users see it without space. If there is a copy and paste code in the first place, shouldn't they write some javascript to do that? Also, a bunch of text without spaces is unreadable.

What's wrong with developer typing in an extra `|cut:" "`? It would be more troublesome to type in a template tag that is a few lines long if they wanted to add it. But comparing to removing it, it's just few characters. Also, it is not possible to add spacing every 4 characters using only CSS from what I know.

Reply:
HTML:
```
<div id="secret-key"><span>0123</span><span>4567</span><span>890a</span><span>bcde</span></div>
```

CSS:

```
#secret-key span {
  padding-left: 0.5em;
  padding-right: 0.5em;
}
```

A working example: https://jsfiddle.net/3x9yj8ao/1/

I don't think it would be too much effort to write a template tag that would output the required HTML for a given key.

That would meet both our requirements:

1. There are visual spaces to make the key easier to read
2. There are no literal spaces, so the key can easily be copied and pasted

Reply:
It is not much effort but compared to adding `cut:" "`, it requires significantly more effort. Or maybe I include the template tag as well? Also, even with literal spaces, it could be copy pasted. I did copy paste it as well, including the space and excluding the space does not matter.